buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1611,18 +1611,48 @@ class BeamModulePlugin implements Plugin<Project> {
         outputs.dirs(project.ext.envdir)
       }
 
+      def pythonSdkDeps = project.files(
+              project.fileTree(dir: "${project.rootDir}", include: [
+                'model/**',
+                'sdks/python/apache_beam/**/*.py',
+                'sdks/python/apache_beam/**/*.pyx',
+                'sdks/python/apache_beam/**/*.pxd',
+                'sdks/python/apache_beam/testing/data/**',
+                'sdks/python/apache_beam/scripts/**',
+                'sdks/python/.pylintrc',
+                'sdks/python/MANIFEST.in',
+                'sdks/python/gen_protos.py',
+                'sdks/python/setup.cfg',
+                'sdks/python/setup.py',
+                'sdks/python/test_config.py',
+                'sdks/python/tox.ini',
+              ])
+              )
+
       project.configurations { distConfig }
 
       project.task('sdist', dependsOn: 'setupVirtualenv') {
         doLast {
+          // Copy sdk sources to isolate directory
+          def copiedSrcDir = "${project.buildDir}/srcs"
+          project.copy {
+            from pythonSdkDeps
+            into copiedSrcDir
+          }
+
+          // Build artifact
           project.exec {
             executable 'sh'
-            args '-c', ". ${project.ext.envdir}/bin/activate && cd ${pythonRootDir} && python setup.py sdist --keep-temp --formats zip,gztar --dist-dir ${project.buildDir}"
+            args '-c', ". ${project.ext.envdir}/bin/activate && cd ${copiedSrcDir}/sdks/python && python setup.py sdist --formats zip,gztar --dist-dir ${project.buildDir}"
```

Comment:
Since we start to use copy of sdk sources to build artifact, there is no need to keep temp directory around after build. So remove `--keep-temp` flag here.

Reply:
For my education, why did we need it before?

Reply:
I added this flag to fix the parallel failure in integration tests since the temp directory is shared between different build processes and by default it's deleted after a build finish. However, it never works for tox tests.